packages/api/src/routes/medical/document.ts

```diff
@@ -134,4 +140,52 @@ router.get(
   })
 );
 
+/**
+ * GET /document/upload-url
+ *
+ * Returns a signed url to upload a file to S3.
+ *
+ * @param patientId - The patientId of the patient.
+ * @param organizationName - The name of the organization that created the document.
+ * @param practitionerName - The name of the practitioner that created the document.
+ * @param fileDescription - The description of the file.
+ *
+ * @return presigned url
+ */
+router.get(
+  "/upload-url",
+  asyncHandler(async (req: Request, res: Response) => {
+    const cxId = getCxIdOrFail(req);
+    const patientId = getFromQueryOrFail("patientId", req);
+    const queryParams = buildQueryParams(req);
+    const docRefId = uuidv7();
+    const s3FileName = createS3FileName(cxId, patientId, docRefId);
+    const presignedUrl = s3client.createPresignedPost({
+      // Bucket: medicalDocumentsUploadBucketName,
+      Bucket: "medical-doc-upload-staging",
+      Fields: {
+        key: s3FileName + "_upload" + queryParams,
+      },
+      Conditions: [
+        ["content-length-range", 0, 25_000_000], // content length restrictions: 0-25MB
+        // ["starts-with", "$Content-Type", "image/"], // content type restriction
+        ["starts-with", "$Content-Type", ""], // content type restriction
+      ],
+    });
+    return res.status(httpStatus.OK).json(presignedUrl);
+  })
+);
+
+function buildQueryParams(req: Request): string {
+  const organizationName = getFromQuery("organizationName", req);
+  const practitionerName = getFromQuery("practitionerName", req);
```

Comment:
And, if so, comes the question about how to provide those since this is currently sending those as query parameter. Could we create the doc ref here on a state that's temporary/draft until the file is uploaded and picked up by the lambda, at which time we'd update the doc ref w/ the final doc location and move the doc ref to "active"?